nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBQueryBuilder.java

```diff
@@ -970,6 +971,16 @@ protected FieldInfo walkReference(String name) {
             if (prop.startsWith(NXQL.ECM_ACL + "/")) {
                 return parseACP(prop, parts);
             }
+            if (prop.startsWith(NXQL.ECM_TAG)) {
```

Comment:
Does this condition is needed ? It seems we already replaced `ECM_TAG` by the new key in `DBSSession`.

Reply:
The replacement is done for the select clause, but we still need to parse the where clause if it contains the `ecm:tag`. As there is a special behaviour if present in the where clause, it was easier to make the replacement here rather than parsing also the `WhereClause` in `DBSSession`.